IHP/FrameworkConfig.hs

```diff
@@ -83,18 +85,29 @@ ihpDefaultConfig = do
     port <- liftIO defaultAppPort
     option $ AppPort port
 
+
     environment <- findOption @Environment
+
+    maybeLogger <- findOptionMaybe @Logger
+    logger <- liftIO $ case maybeLogger of
+        Just logger -> pure logger
+        Nothing -> case environment of
+            Development -> defaultLogger
+            Production -> newLogger (def { level = Warn } :: LoggerSettings)
+    option logger
```

Comment:
Actually the `option` function doesn't override a value if it has been set previously.

So e.g.

```haskell
option $
    case environment of
            Development -> defaultLogger
            Production -> newLogger (def { level = Warn } :: LoggerSettings)
```

would be equivalent to the code above